### PR TITLE
Fixing aliases for GLOBAL IN

### LINKED ISSUE #64445: Logical Error: Expression nodes list expected 1 projection names. Actual 2.
**Describe the bug**
A [link](https://s3.amazonaws.com/clickhouse-test-reports/64311/0383fa5164cb07fdec7c5fc036137122545acd6a/ast_fuzzer__debug_.html) to the report

**How to reproduce**
```sql
SELECT 1 GLOBAL IN (SELECT 1) AS s, s FROM remote('127.0.0.{2,3}', system.one) GROUP BY 1
```
```
Received exception from server (version 24.6.1):
Code: 49. DB::Exception: Received from localhost:9000. DB::Exception: Received from 127.0.0.2:9000. DB::Exception: Expression nodes list expected 1 projection names. Actual 2. (LOGICAL_ERROR)
```

### PR DESCRIPTION
**Pull request by KochetovNicolai**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix the `Expression nodes list expected 1 projection names` and `Unknown expression or identifier` errors for queries with aliases to `GLOBAL IN.`

Fixes #64445

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 506bc44cbe99d5f663e6db093b911c8f03cdda91 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64517/506bc44cbe99d5f663e6db093b911c8f03cdda91/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://github.com/search?q=head%3Async-upstream%2Fpr%2F64517+org%3AClickHouse+type%3Apr">⏳ pending</a></td></tr>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/64517/506bc44cbe99d5f663e6db093b911c8f03cdda91/ci_running.html">⏳ pending</a></td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/64428/8166da7fbb616d9fa2d779ffe8e533b238d3680e/docs_check.html">✅ success</a></td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/64517/506bc44cbe99d5f663e6db093b911c8f03cdda91/fast_test.html">✅ success</a></td></tr>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/64517/506bc44cbe99d5f663e6db093b911c8f03cdda91/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Review by novikd — COMMENTED** [review]
Please, add a test from the issue description.

File: src/Analyzer/Passes/QueryAnalysisPass.cpp [hunk 1]
```diff
@@ -6566,7 +6578,8 @@ ProjectionNames QueryAnalyzer::resolveExpressionNode(QueryTreeNodePtr & node, Id
                     scope.scope_node->formatASTForErrorMessage());
 
             auto & table_node = node->as<TableNode &>();
-            result_projection_names.push_back(table_node.getStorageID().getFullNameNotQuoted());
+            if (result_projection_names.empty())
```
Comment (novikd) [inline src/Analyzer/Passes/QueryAnalysisPass.cpp:6581]: Just curious: when can it be non-empty?

**Review by novikd — APPROVED** [review]

File: src/Analyzer/Passes/QueryAnalysisPass.cpp (hunk 1 above)
Comment (KochetovNicolai) [reply src/Analyzer/Passes/QueryAnalysisPass.cpp:6581]: In the scenario I saw, we can resolve identifier `x alias y` as TableNode (by alias), and in this case, at first, alias would be added to projection names, and then the storage name.

**Comment by alexey-milovidov** [comment]
Cloud does not use GLOBAL IN; backport still makes sense to simplify other backports.

**Comment by KochetovNicolai** [comment]
In stress tests, probably we stuck in distributed async inserts.
```
| | | | | | | | | | | | | | | | | | | | | | | | | | | 14   0x000000001c9aafdf in DB::DistributedAsyncInsertDirectoryQueue::run at /build/src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp:200
| | | | | | | | | | | | | | | | | | | | | | | | | | | | 13   0x000000001c9aab26 in DB::DistributedAsyncInsertDirectoryQueue::processFiles at /build/src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp:374
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x000000001c9af031 in DB::DistributedAsyncInsertDirectoryQueue::processFile at /build/src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp:430
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x000000001c9c1fe8 in DB::RemoteInserter::onFinish at /build/src/QueryPipeline/RemoteInserter.cpp:127
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x000000001d23ef4b in DB::Connection::receivePacket at /build/src/Client/Connection.cpp:1074
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x0000000012d7c0ac in DB::readVarUInt at /build/src/IO/VarInt.h:114
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   DB::varint_impl::readVarUInt<true> at /build/src/IO/VarInt.h:95
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x0000000012d7c34f in DB::ReadBuffer::eof at /build/src/IO/ReadBuffer.h:106
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x000000000afe6a0a in DB::ReadBuffer::next at /build/src/IO/ReadBuffer.h:70
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x0000000012f03b6a in DB::ReadBufferFromPocoSocket::nextImpl at /build/src/IO/ReadBufferFromPocoSocket.cpp:81
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x0000000022778ead in Poco::Net::SecureStreamSocketImpl::receiveBytes at /build/base/poco/NetSSL_OpenSSL/src/SecureStreamSocketImpl.cpp:139
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x00000000227752fd in Poco::Net::SecureSocketImpl::receiveBytes at /build/base/poco/NetSSL_OpenSSL/src/SecureSocketImpl.cpp:350
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x0000000022773e7f in Poco::Net::SecureSocketImpl::mustRetry at /build/base/poco/NetSSL_OpenSSL/src/SecureSocketImpl.cpp:480
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x00000000227b8b88 in Poco::Net::SocketImpl::pollImpl at /build/base/poco/Net/src/SocketImpl.cpp:436
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 12   0x00007fb0e76c8bcf in poll () from /lib/x86_64-linux-gnu/libc.so.6
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x000000001c9aef76 in DB::DistributedAsyncInsertDirectoryQueue::processFile at /build/src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp:424
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x000000001c9c1918 in DB::RemoteInserter::RemoteInserter at /build/src/QueryPipeline/RemoteInserter.cpp:59
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x000000001d23ef4b in DB::Connection::receivePacket at /build/src/Client/Connection.cpp:1074
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x0000000012d7c0ac in DB::readVarUInt at /build/src/IO/VarInt.h:114
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    DB::varint_impl::readVarUInt<true> at /build/src/IO/VarInt.h:95
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x0000000012d7c34f in DB::ReadBuffer::eof at /build/src/IO/ReadBuffer.h:106
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x000000000afe6a0a in DB::ReadBuffer::next at /build/src/IO/ReadBuffer.h:70
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x0000000012f03b6a in DB::ReadBufferFromPocoSocket::nextImpl at /build/src/IO/ReadBufferFromPocoSocket.cpp:81
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x0000000022778ead in Poco::Net::SecureStreamSocketImpl::receiveBytes at /build/base/poco/NetSSL_OpenSSL/src/SecureStreamSocketImpl.cpp:139
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x00000000227752fd in Poco::Net::SecureSocketImpl::receiveBytes at /build/base/poco/NetSSL_OpenSSL/src/SecureSocketImpl.cpp:350
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x0000000022773e7f in Poco::Net::SecureSocketImpl::mustRetry at /build/base/poco/NetSSL_OpenSSL/src/SecureSocketImpl.cpp:480
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x00000000227b8b88 in Poco::Net::SocketImpl::pollImpl at /build/base/poco/Net/src/SocketImpl.cpp:436
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x00007fb0e76c8bcf in poll () from /lib/x86_64-linux-gnu/libc.so.6
| | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x000000001c9aabbc in DB::DistributedAsyncInsertDirectoryQueue::processFiles at /build/src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp:377
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x000000001c9af031 in DB::DistributedAsyncInsertDirectoryQueue::processFile at /build/src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp:430
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x000000001c9c1fe8 in DB::RemoteInserter::onFinish at /build/src/QueryPipeline/RemoteInserter.cpp:127
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x000000001d23ef4b in DB::Connection::receivePacket at /build/src/Client/Connection.cpp:1074
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x0000000012d7c0ac in DB::readVarUInt at /build/src/IO/VarInt.h:114
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    DB::varint_impl::readVarUInt<true> at /build/src/IO/VarInt.h:95
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x0000000012d7c34f in DB::ReadBuffer::eof at /build/src/IO/ReadBuffer.h:106
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x000000000afe6a0a in DB::ReadBuffer::next at /build/src/IO/ReadBuffer.h:70
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x0000000012f03b6a in DB::ReadBufferFromPocoSocket::nextImpl at /build/src/IO/ReadBufferFromPocoSocket.cpp:81
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x0000000022778ead in Poco::Net::SecureStreamSocketImpl::receiveBytes at /build/base/poco/NetSSL_OpenSSL/src/SecureStreamSocketImpl.cpp:139
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x00000000227752fd in Poco::Net::SecureSocketImpl::receiveBytes at /build/base/poco/NetSSL_OpenSSL/src/SecureSocketImpl.cpp:350
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x0000000022773e7f in Poco::Net::SecureSocketImpl::mustRetry at /build/base/poco/NetSSL_OpenSSL/src/SecureSocketImpl.cpp:480
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x00000000227b8b88 in Poco::Net::SocketImpl::pollImpl at /build/base/poco/Net/src/SocketImpl.cpp:436
| | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | 1    0x00007fb0e76c8bcf in poll () from /lib/x86_64-linux-gnu/libc.so.6
```

**Pull request merged**